internal/states/statemgr/locker.go

```diff
@@ -140,6 +143,9 @@ type LockInfo struct {
 
 	// Path to the state file when applicable. Set by the Lock implementation.
 	Path string
+
+	// Set when there is race conditions on locking in non-locking backends (GCS)
```

Comment:
What do you mean by "non-locking backends"? GCS is a locking backend.
I think it would also make sense to explain what the exact race condition is (write fails because of conflict, read returns nothing).

Reply:
updated with 
```
	// Set when writing of lock file fails because of conflict and
	// then reading fails because file doesn't exist (removed by other process)
```